server/hotrod/src/main/java/org/infinispan/server/hotrod/LifecycleCallbacks.java

```diff
@@ -80,12 +106,9 @@ private void registerServerTransactionTable(ComponentRegistry componentRegistry,
       }
       EmbeddedCacheManager cacheManager = componentRegistry.getGlobalComponentRegistry()
             .getComponent(EmbeddedCacheManager.class);
-      Cache<CacheXid, TxState> globalTxTable = cacheManager.getCache(GLOBAL_TX_TABLE_CACHE_NAME);
-      ServerTransactionTable transactionTable = new ServerTransactionTable(globalTxTable,
-            ByteString.fromString(cacheName), cacheManager.getAddress());
-      componentRegistry.registerComponent(transactionTable, ServerTransactionTable.class);
-      ServerTransactionOriginatorChecker checker = new ServerTransactionOriginatorChecker();
-      componentRegistry.registerComponent(checker, TransactionOriginatorChecker.class);
+      createGlobalTxTable(cacheManager);
+      componentRegistry.registerComponent(new PerCacheTxTable(cacheManager.getAddress()), PerCacheTxTable.class);
+      componentRegistry.registerComponent(new ServerTransactionOriginatorChecker(), TransactionOriginatorChecker.class);
       componentRegistry.rewire();
```

Comment:
Could other core components depend on the components we've just added? Otherwise `rewire()` doesn't do anything.

Reply:
at least the TransactionTable in core needs the `TransactionOriginatorChecker`.

Reply:
it used in hibernate cache to replace some components. I wasn't aware of that!
a flag in GlobalTransaction will eliminate the use of ServerTransactionOriginatorChecker. 
I actually would like to replace the GlobalTransaction to an interface and have different impls for embedded and remote...